### ISSUE TITLE
a11y issue: Tree item at level 2 is not announced correctly by screen reader.

### ISSUE THREAD

**Issue by lateefsofi** [SAP/ui5-webcomponents#2553]
Tree item at level 2 is not announced correctly by screen reader.
Consider [Basic Tree](https://sap.github.io/ui5-webcomponents/playground/components/Tree/) to understand the issue.
When screen reader user navigates through the tree at level 2 i.e _Tree 1.1_ as per example , screen reader announces 1 of 3 but actually it should announce 1 of 1, because there is only 1 item under its parent.
Similarly if you navigate to the level 2 under _Tree 2_, for the first item screen reader announces 2 of 3 which is again incorrect, it should be 1 of 2.

**To reproduce**
1. Go to https://sap.github.io/ui5-webcomponents/playground/components/Tree
2. When screen reader user navigates through the tree at level 2 i.e _Tree 1.1_ as per example , screen reader announces 1 of 3 but actually it should announce 1 of 1, because there is only 1 item under its parent.
Similarly if you navigate to the level 2 under _Tree 2_, for the first item screen reader announces 2 of 3 which is again incorrect, it should be 1 of 2.

**Expected behavior**
Items should be announced to screen readers correctly.

**Screenshots**
![image](https://user-images.githubusercontent.com/18447407/101469399-b9b69780-396a-11eb-9499-00952be5fc60.png)

**Context**
- UI5 Web Components version: 0.23.6

**Affected components** *(if known)* Tree

**Organization:** *(if applicable)* SuccessFactors

**Priority:** *(optional)* *(Low, Medium, High or Very High)* High

**Comment by NHristov-sap** [comment]
Hi,

The issue is reproducible. Sometimes (for level 0) it reads properly, but the problem is when the whole tree is expanded, especially in Tree 2 and its sub-trees (when expanded). There are correct announcements of the Tree nodes, but incorrect announcements for number of items (X of Y). Please check!

Best Regards,
Nikolay Hristov
(GitHub Dispatcher)
